scripts/pylib/twister/twisterlib.py

```diff
@@ -2960,7 +2960,9 @@ def add_testcases(self, testcase_filter=[]):
                         if tc.harness == 'console' and not tc.harness_config:
                             raise Exception('Harness config error: console harness defined without a configuration.')
                         tc.build_only = tc_dict["build_only"]
-                        tc.build_on_all = tc_dict["build_on_all"]
+
```

Comment:
Shouldn't we add any warning or message?

Reply:
I don't think so. I am pretty sure 99% of sample/testcase.yamls were created as a copy passed based on some wrong examples and some "bad habits" spread. In this case, the problem is with "build_on_all: true". As I pointed out, it makes no sense in any of nrf specific samples/tests/apps. There is no way that any of our tests/samples/apps can be build on all ~400 platforms that exist in zephyr.  Using "build_on_all: true" is clearly a misuse in nrf yamls because there is also platform_allow used. These entries are contradictory.
I am think of creating later on some script that will automatically remove all this  "build_on_all: true" lines from nrf yamls.

Reply:
I think removing all the 'build_on_all' lines in sdk-nrf will be faster than applying this temp-hack, doing manifest update in sdk-nrf, and then rolling it back later.

Reply:
making a script removing all the 'build_on_all' lines in sdk-nrf samples/tests/apps yamls is not hard but needs a bit more caution.  It will also require a PR to sdk-nrf which will modify all these files. If you think such temp hack is not the best idea I will start working on the script.

Reply:
either way is fine for me, but making two changes in zephyr will be slow, and the current upmerge might cause a problem.